### PR TITLE
[PW-5446] Now capturing amount in correct currency when setting a different display currency

### LINKED ISSUE #1138: [PW-5446] Incorrect capture amount in payment request
Hi, 
there is a problem with the capture when the Charged currency is set at the "display" level.

I have Adyen module to version 7.0.1 and to support the multicapture I added the patch[ [PW-5005] Capture subject amount instead of order full amount #1112](https://github.com/Adyen/adyen-magento2/pull/1112) 
In this case, the invoice on Magento won't capture the correct amount.

I found this problem in my environment, where an order was placed on a website with £239.11 as grand total and on the admin it has €281.13 as a base grand total.

When I performed the capture, there is mixed information added in the capture request:

```
(
[currency] => GBP
[value] => 28113
)

```
The right value should be 23911.

In this case, the capture didn't work because the authorized amount was less than the captured one, and I received this message as notification: "reason: Insufficient balance on payment"
But with a different currency, where the exchange rate is better for the EURO, I will capture a really small amount:
base grand total: €461.04 
grand total: CN¥3,507.17

capture request:
```
(
[currency] => CNY
[value] => 46104
)
```
 I lost ~3100 CNY

**To Reproduce**
1. enable different currencies on Magento
2. set, inside the config.php file, the display value for charged_currency in a website 
 ```
... 
       'payment' => [
                    'adyen_abstract' => [
                        'charged_currency' => 'display'
                    ]
         ],
...
 ```
3. set Adyen capture as manual
4. on the frontend, view the website in a currency different from the base currency
5. create an order
6. on the admin, create an invoice
7. wait for the notification 
8. if the exchange rate is unfavourable, you will get an "Insufficient balance on payment" notification message; otherwise, you captured less money for your item 

**Expected behavior**
The captured amount should be check the charged_currency feature

**Magento version**
2.4.1

**Plugin version**
7.0.1 with patch PW-5005

### PR DESCRIPTION
**Pull request by tnaber**
**Description**
Previously the grandtotal in the wrong currency would be selected for the invoice when a different display currency was configured. This is now fixed.
To recreate the scenario, see the linked issue and the corresponding ticket on yt.

**Tested scenarios**
- Tested in 7.2.0 and 7.3.0
- Tested with display currency USD, both when selecting EU and USD
- Tested for partial payments

**Fixed issue**:  
Closes #1138

### CODE REVIEW

File: Gateway/Request/CaptureDataBuilder.php [hunk 1]
```diff
@@ -109,10 +109,12 @@ public function build(array $buildSubject)
     {
         /** @var \Magento\Payment\Gateway\Data\PaymentDataObject $paymentDataObject */
         $paymentDataObject = \Magento\Payment\Gateway\Helper\SubjectReader::readPayment($buildSubject);
-        $amount = \Magento\Payment\Gateway\Helper\SubjectReader::readAmount($buildSubject);
         $payment = $paymentDataObject->getPayment();
         /** @var Order $order */
         $order = $payment->getOrder();
+        $latestInvoice = $order->getInvoiceCollection()->getLastItem();
+        $amount = $latestInvoice->getGrandTotal();
```
[the remark this answers is not in the document]
Comment (tnaber) [reply Gateway/Request/CaptureDataBuilder.php]: Hi @mirkocesaro,

I can not reproduce these errors on my system. Would it be possible for you to further investigate the issue by debugging the 'isTotalAmountAuthorized' function? I suspect something might be going wrong with the authorization notification.

If this does not yield any potential causes, could you send me the following info:
- All info.log logs related to this payment
- Steps for recreating the error

Cheers,
Titus

**Review by michaelpaul — APPROVED** [review]

File: Gateway/Request/CaptureDataBuilder.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (tnaber) [reply Gateway/Request/CaptureDataBuilder.php]: Hi @mirkocesaro,

As we feel like the issue from #1197 is fixed by this PR we have decided to go ahead with the merge. Please feel free to open a new issue (with the extra info) for your current problem :)

Cheers,
Titus

**Comment by dufrygrant** [comment]
Is there a reason why this wasn't patched into 7,x branch as a priority (for production users of that branch) over its inclusion in 8.0.0 branch?

**Comment by tnaber** [comment]
Hi @dufrygrant, we will be releasing a version of v7 with this patch implemented :)

**Pull request merged**